### PR TITLE
Replace instances of `100 - score < 0` -> Math.max

### PR DESCRIPTION
**Pull request by keithamus**
In some rules, the `result.score` key was using a ternary operator to detect if
`100 - score` was `< 0`, and if it was, then the property would be set to 0.

A better way to handle this is Math.max:

```js
score = 80;
(100 - score < 0 ? 0 : 100 - score) === 20
Math.max(0, 100 - score) === 20

score = 150
(100 - score < 0 ? 0 : 100 - score) === 0
Math.max(0, 100 - score) === 0
```

### CODE REVIEW

**Comment by tobli** [comment]
That's a great improvement. If you don't mind, we'd happily accept a PR that applies the same Math.max logic applied to all advice in one commit. That way the code base stays consistent.

Also if you rebase against master you'll get an updated .travis.yml. We've kept the dist folder checked in, and had a check in Travis that commits included updated minified scripts. Terrible idea, and we've just now changed that. Now Travis should accept your PR.

Thanks for contributing to open source!
/Tobias

**Comment by keithamus** [comment]
Updated to catch all instances.

**Comment by keithamus** [comment]
Of course, as it is a pattern that is repeated so much - perhaps it would be better to handle upstream in the result fetcher? So if a result returns score < 0 then it is bound to 0?

File: lib/har/performance/assetsRedirects.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ module.exports = {
     });
 
     return {
-      score: score < 0 ? 0 : score,
```
Comment (tobli) [inline lib/har/performance/assetsRedirects.js]: No 100 - score here.

File: lib/har/performance/cacheHeaders.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ module.exports = {
     });
 
     return {
-      score: score < 0 ? 0 : score,
```
Comment (tobli) [inline lib/har/performance/cacheHeaders.js]: or here

File: lib/har/performance/compressAssets.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ module.exports = {
     });
 
     return {
-      score: score < 0 ? 0 : score,
```
Comment (tobli) [inline lib/har/performance/compressAssets.js]: ditto

**Comment by tobli** [comment]
Some places differ slightly in their logic, so that score is deducted from a starting point of 100. As you wrote, there's likely better way to make this better and more consistent. I propose fixing the test failures and applying this PR, and see if we can simplify/unify further as a follow up. Thanks again!

**Comment by keithamus** [comment]
Oh god... sorry I screwed up the find and replace! Fixed now.

Shall I follow this up as a separate PR?

**Comment by tobli** [comment]
Merging this now. Follow up PRs for further refinements are would be great.

**Pull request merged**